src/WebJobs.Script.WebHost/WebScriptHostRequestManager.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Binding.Http;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class WebScriptHostRequestManager : HttpRequestManager
+    {
+        private readonly HostPerformanceManager _performanceManager;
+        private readonly IMetricsLogger _metricsLogger;
+        private readonly int _performanceCheckPeriodSeconds;
+        private DateTime _lastPerformanceCheck;
+        private bool _rejectAllRequests;
+
+        public WebScriptHostRequestManager(HttpConfiguration config, HostPerformanceManager performanceManager, IMetricsLogger metricsLogger, TraceWriter traceWriter, int performanceCheckPeriodSeconds = 15) : base(config, traceWriter)
+        {
+            _performanceManager = performanceManager;
+            _metricsLogger = metricsLogger;
+            _performanceCheckPeriodSeconds = performanceCheckPeriodSeconds;
+        }
+
+        protected override bool RejectAllRequests()
```

Comment:
By popular consensus, I'm making this configurable, on by default. It can be disabled by setting a new "dynamicThrottlesEnabled" setting to false.